Source/SwiftLintBuiltInRules/Rules/Lint/PrivateSwiftUIStatePropertyRule.swift

```diff
@@ -0,0 +1,108 @@
+import SwiftSyntax
+
+/// Require that any state properties in SwiftUI be declared as private
+///
+/// State properties should only be accessible from inside the View's body, or from methods called by it
+struct PrivateSwiftUIStatePropertyRule: SwiftSyntaxRule, OptInRule, ConfigurationProviderRule {
+    var configuration = SeverityConfiguration<Self>(.warning)
+
+    static let description = RuleDescription(
+        identifier: "private_swiftui_state",
+        name: "Private SwiftUI @State Properties",
+        description: "SwiftUI's state properties should be private",
+        kind: .lint,
+        nonTriggeringExamples: [
+            Example("""
+            struct ContentView: View {
+                @State private var isPlaying: Bool = false
+            }
+            """),
+            Example("""
+            struct ContentView: View {
+                @State fileprivate var isPlaying: Bool = false
+            }
+            """),
+            Example("""
+            struct ContentView: View {
+                var isPlaying = false
+            }
+            """),
+            Example("""
+            struct ContentView: View {
+                @StateObject var foo = Foo()
+            }
+            """),
+            Example("""
+            struct Foo {
+                @State var bar = false
+            }
+            """),
+            Example("""
+            class Foo: ObservableObject {
+                @State var bar = Bar()
+            }
+            """)
+        ],
+        triggeringExamples: [
+            Example("""
+            struct ContentView: View {
+                @State ↓var isPlaying: Bool = false
+            }
+            """
+            )
+        ]
+    )
+
+    func makeVisitor(file: SwiftLintFile) -> ViolationsSyntaxVisitor {
+        Visitor(viewMode: .sourceAccurate)
+    }
+}
+
+private extension PrivateSwiftUIStatePropertyRule {
+    final class Visitor: ViolationsSyntaxVisitor {
+        override var skippableDeclarations: [DeclSyntaxProtocol.Type] {
+            .allExcept(StructDeclSyntax.self)
```

Comment:
Final classes can conform to `View` as well if I remember correctly.

Other than that, we need to consider structs in structs and structs in classes and ... as well. Hence, it'd be better to exclude certain types instead of accepting structs exclusively.

Reply:
👍 I'll make a change. Do you think that skipping extensions and protocol declarations might be sufficient here? Neither of those declarations supported nested classes, so that might cover any of this rule's cases.

Reply:
Never mind, I don't think that's a good assumption on my part either. I'm going to update the visitor accordingly and get a commit up in a bit.